### PR TITLE
[ws_pcan_js] Add Suspicious XML comment disclosure

### PR DESCRIPTION
**Pull request by eakirtas**

### CODE REVIEW

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/suspicious_xml_comments_disclosure.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
+// This Script will seek at incoming WebSocket XML formatted messages for suspicious comments.
+
+// * Is based on: org.zaproxy.zap.extension.pscanrulesBeta.InformationDisclosureSuspiciousComments
+// ** And the comment list is copied by
```
Comment (kingthorin) [inline addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/suspicious_xml_comments_disclosure.js]: "by" > "from"

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/suspicious_xml_comments_disclosure.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    commentsList.forEach(function(comment){
+        commentPatterns.forEach(function(pattern){
+            if(pattern.test(comment)){
+                helper.newAlert()
+                    .setRiskConfidence(RISK_INFO, CONFIDENCE_MEDIUM)
+                    .setName("Information Disclosure - Suspicious Comments in XML via WebSocket (script)")
+                    .setDescription("The response appears to contain suspicious comments which may help an attacker.")
+                    .setParam(counter++)
+                    .setSolution("The response appears to contain suspicious comments which may help an attacker.")
```
Comment (kingthorin) [inline addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/suspicious_xml_comments_disclosure.js]: This isn't a solution...

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/suspicious_xml_comments_disclosure.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                    .setWascId(13) //WASC Id 13 - Info Leakage
+                    .raise();
+            }
+        });
+    });
+}
+
+function getComments(node, commentsList){
+	  var nodeList = node.getChildNodes();
+    for(var i = 0; i < nodeList.getLength(); i++){
```
Comment (kingthorin) [inline addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/suspicious_xml_comments_disclosure.js]: There seems to be some mixed indentation here (this line vs. previous).

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/suspicious_xml_comments_disclosure.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    getComments(xmlDoc.getDocumentElement(), commentsList);
+
+    commentsList.forEach(function(comment){
+        commentPatterns.forEach(function(pattern){
+            if(pattern.test(comment)){
+                helper.newAlert()
+                    .setRiskConfidence(RISK_INFO, CONFIDENCE_MEDIUM)
+                    .setName("Information Disclosure - Suspicious Comments in XML via WebSocket (script)")
+                    .setDescription("The response appears to contain suspicious comments which may help an attacker.")
+                    .setParam(counter++)
```
Comment (kingthorin) [inline addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/suspicious_xml_comments_disclosure.js]: I think this is probably used on some handling in the [Find HTML Comments](https://github.com/zaproxy/community-scripts/blob/27efdf1cb39dee4f9dd71d7055e4b5a6ee231c42/passive/Find%20HTML%20Comments.js#L23-L24) script. However: 1) It's an older script. 2) It uses parameters in order to facilitate different grouping of results.

In the majority of cases passive scanners (scripts or other) should not really have/need a "parameter".

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/suspicious_xml_comments_disclosure.js (hunk 1 above)
Comment (eakirtas) [reply addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/suspicious_xml_comments_disclosure.js]: Fixed in fdcd0adac

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/suspicious_xml_comments_disclosure.js (hunk 2 above)
Comment (eakirtas) [reply addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/suspicious_xml_comments_disclosure.js]: Fixed in fdcd0adac

File: addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/suspicious_xml_comments_disclosure.js (hunk 4 above)
Comment (eakirtas) [reply addOns/websocket/src/main/zapHomeFiles/scripts/scripts/websocketpassive/suspicious_xml_comments_disclosure.js]: Fixed in 42d6098cb

**Pull request closed without merging**